src/components/modals/AddMovieModal/AddMovieModal.js

```diff
@@ -15,89 +19,147 @@ const FIELDS = [
     readOnly: true,
     key: 'id',
   },
-  { label: 'Title', type: 'text', key: 'title' },
-  { label: 'Release Date', type: 'date', key: 'release_date' },
-  { label: 'Poster Url', type: 'text', key: 'poster_path' },
+  {
+    label: 'Title',
+    type: 'text',
+    key: 'title',
+    placeholder: 'Input title',
+    validationSchema: Yup.string().required().label('Title'),
+  },
+  {
+    label: 'Release Date',
+    type: 'date',
+    key: 'release_date',
+    placeholder: 'Select Date',
+    validationSchema: Yup.date().nullable().label('Release Date'),
+  },
+  {
+    label: 'Poster Url',
+    type: 'text',
+    key: 'poster_path',
+    placeholder: 'Poster URL here',
+    validationSchema: Yup.string().url().required()
+      .label('Poster Url'),
+  },
   {
     label: 'Genre',
     type: 'select',
     values: GENRES,
     key: 'genres',
+    placeholder: 'Select Genre(s)',
+    validationSchema: Yup.array()
+      .of(Yup.string().oneOf(GENRES))
+      .min(1)
+      .nullable()
+      .label('Genres'),
+  },
+  {
+    label: 'Runtime',
+    type: 'number',
+    key: 'runtime',
+    placeholder: 'Runtime here',
+    validationSchema: Yup.number()
+      .required()
+      .positive()
+      .integer()
+      .label('Runtime'),
+  },
+  {
+    label: 'Overview',
+    type: 'textarea',
+    key: 'overview',
+    placeholder: 'Overview here',
+    validationSchema: Yup.string().required().label('Overview'),
   },
-  { label: 'Runtime', type: 'number', key: 'runtime' },
-  { label: 'Overview', type: 'textarea', key: 'overview' },
 ];
 
 const getOptions = (options) => (options ? options.map((option) => ({ label: option, value: option })) : []);
 
 const GENRES_OPTIONS = getOptions(GENRES.slice(0).sort(sortAbc));
 
-const AddMovieModal = ({
-  isEdit, movie, onAction, show,
-}) => {
-  const [mutableMovie, setMutableMovie] = useState({ ...movie });
+const validationSchemas = Object.assign(
+  {},
+  ...FIELDS.filter((field) => field.validationSchema).map((field) => ({
+    [field.key]: field.validationSchema,
+  })),
+);
```

Comment:
```js
const validationSchemas = {
  ...FIELDS.filter((field) => field.validationSchema).map(({key, validationSchema}) => ({
    [key]: validationSchema,
  })),
}
```

Reply:
Your solution won't work because it will return an object kind of `{0: {...}, 1: {...}, 2: {...}, ...}` while I need an object created as a union of objects from array. Anyway, I rewrote this part using `reduce`